### DISCUSSION TITLE
Response cookies not working

### DISCUSSION THREAD

**Discussion by cldtech** [fastapi/fastapi#10536] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@auth.post('/register/step1')
def register_step1(step1: RegStep1):
    content = {"status": "success"}
    response = JSONResponse(content=content)
    response.set_cookie(key="registration_id", value=uuid)
    return response
```

### Description

According to [the doc](https://fastapi.tiangolo.com/advanced/response-cookies/) sending a response cookie if pretty simple exepct it's not working at all. I tried the exact example of the doc and it doesn't add the set cookie header.
I went in the code to see what's not happening but and i can't even find the set_cookie function in the JSONResponse object. What's happening with this feature? Thanks.

### Operating System

Linux

### FastAPI Version

0.103.1

### Pydantic Version

2.3.0

### Python Version

3.9.2

**Comment by jgould22** [comment]
The set cookie function is [here](https://github.com/encode/starlette/blob/master/starlette/responses.py#L86) 

How do you know the cookie is not set?

When I run the sample code in the docs I see it.

```bash
$ curl -I -X POST http://127.0.0.1:8000/cookie/
HTTP/1.1 200 OK
date: Fri, 27 Oct 2023 17:52:25 GMT
server: uvicorn
content-length: 52
content-type: application/json
set-cookie: fakesession=fake-cookie-session-value; Path=/; SameSite=lax
```

**Reply by cldtech** [reply]
Well i don't receive it on the client side looking from firefox developper tools (storage tab), or even from a simple api tester the response i receive is this;
![Screenshot from 2023-10-27 14-12-38](https://github.com/tiangolo/fastapi/assets/149193495/d089bd5e-6576-4d65-a72a-bd07c4fa107f)

**Reply by jgould22** [reply]
Yes, the swagger UI cannot access that header, so it wont appear in the list.

You can see the warning on the top of this page.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

**Reply by cldtech** [reply]
Okay, but the cookie still doesn't end up in my front-end, i should see it in dev tools. The fetch request succeed but i don't receive the cookie.

**Reply by jgould22** [reply]
Using the example in the documentation the cookies appears in the response, and is sent with subsequent requests 
![image](https://github.com/tiangolo/fastapi/assets/4335847/bd49ff8a-ca27-48bd-afd2-276359e9c2d0)

What does your request look like in the dev tools?

**Reply by cldtech** [reply]
Okay, after purging my venv and reinstalling fastapi some how it works. I now see the set-cookie header sent by the server, but it still doesn't show in the storage tab and is not sent back to the server in the next request header. I guess it's not a fastapi problem anymore.
![Screenshot from 2023-10-27 17-49-21](https://github.com/tiangolo/fastapi/assets/149193495/0785b19f-9aae-46ba-b7f5-9691fbb0fec9)
